src/content/docs/en/guides/deploy/space.mdx

```diff
@@ -0,0 +1,122 @@
+
+This guide includes step-by-step instructions for building sites in Space. You can build any type of Astro site in Space: either a static site or a server-rendered site.
+
+## Prerequisite
+
+To push an Astro site to Space, make sure you first:
+
+- Create a [Space account](https://deta.space)
+- Install the [Space CLI](https://deta.space/docs/en/basics/cli) and log in
+
+No matter the type of Astro site you are building, create a Space project **inside** the directory of your Astro project. Run the CLI and follow the instructions on the screen.    
+
+```bash
+space new
+```
+
+:::note
+The Space CLI will try to detect the right configuration for your app. Accept the suggested configuration and follow the instructions below on how to configure it for your Astro app.
+:::
+
+
+## Project Configuration
+
+### Static site
+
+If you build a simple Astro site with no adapters, you don’t need any extra configuration to your Astro project. 
+
+Edit the `Spacefile` file generated by the Space CLI on the root of your project. You will need to change the engine to `static`, add the `build` command, and serve the `dist` directory generated by Astro. 
+
+```yaml title="Spacefile"
+# Spacefile Docs: https://go.deta.dev/docs/spacefile/v0
+v: 0
+micros:
+  - name: static-astro-in-space
+    src: .
+    engine: static
+    commands:
+      - npm run build
+    serve: dist
+```
+
+### Server-side rendering
+
+If you wish to use [server-side rendering (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) in your Astro project, you will need to install the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter first. 
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npx astro add node
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn astro add node
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm astro add node
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+If you need to further configure the adapter, please follow the [`@astrojs/node` documentation](https://docs.astro.build/en/guides/integrations-guide/node/).
+
+:::caution
+When migrating from a static site to a server-rendered site, make sure you make all the proper adjustments based on the type of Astro site you are building. There are bits of configuration sprinkled around the documentation worth checking, but it's not limited to:
+
+- [Dynamic routes for SSR](https://docs.astro.build/en/core-concepts/routing/#server-ssr-mode)
+- [Modifying the `[...slug]` example for SSR](https://docs.astro.build/en/core-concepts/routing/#modifying-the-slug-example-for-ssr)
+- [Building the content collections for SSR](https://docs.astro.build/en/guides/content-collections/#building-for-server-output-ssr)
+:::
```

Comment:
I'm going to also consult with @Jutanium on this one. My thinking is that someone already HAS either an SSG or SSR site, and is just needing to know how to deploy it. I'm not sure someone would actually be converting their site *at this stage of the game*.

But, I love what you wrote about migrating to SSR, and I'm not even sure WE have that in our docs in exactly this way? :smile: We may nix it from here, but actually make this its own recipe!

Reply:
> My thinking is that someone already HAS either an SSG or SSR site, and is just needing to know how to deploy it. I'm not sure someone would actually be converting their site at this stage of the game.

Yess, I am the same here. Got major mixed feelings about this as well (still do). Happy to chime in the conversation if needed 🙌🏼

Reply:
Let's remove this, and move the discussion [here](https://github.com/withastro/docs/issues/2792#issuecomment-1472428182).

```suggestion
If you wish to use [server-side rendering (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) in your Astro project, you will need to install the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter first. 

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  npx astro add node
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  yarn astro add node
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  pnpm astro add node
  ```
  </Fragment>
</PackageManagerTabs>

If you need to further configure the adapter, please follow the [`@astrojs/node` documentation](https://docs.astro.build/en/guides/integrations-guide/node/).
```

Reply:
Also, it might be worth explicitly saying that Space _only_ works with Node. That's inferred here, but it's worth being more clear.